### PR TITLE
CALCITE-5327 Make SSL key-store type configurable

### PR DESCRIPTION
**Pull request by richardantal**

### CODE REVIEW

File: server/src/main/java/org/apache/calcite/avatica/server/HttpServer.java [hunk 1]
```diff
@@ -850,6 +875,9 @@ protected SslContextFactory.Server buildSSLContextFactory() {
         sslFactory.setKeyStorePassword(keystorePassword);
         sslFactory.setTrustStorePath(truststore.getAbsolutePath());
         sslFactory.setTrustStorePassword(truststorePassword);
+        if (keystoreType != null && !keystoreType.equals(DEFAULT_KEYSTORE_TYPE)) {
+          sslFactory.setKeyStoreType(keystoreType);
```
Comment (joshelser) [inline server/src/main/java/org/apache/calcite/avatica/server/HttpServer.java:879]: Could you add a comment here as to why this is important? Was there some use-case which can accept a keystore in a different type?

**Comment by F21** [comment]
+1 to @joshelser's comments, otherwise everything else looks great!

File: server/src/main/java/org/apache/calcite/avatica/server/HttpServer.java (hunk 1 above)
Comment (richardantal) [reply server/src/main/java/org/apache/calcite/avatica/server/HttpServer.java:876]: Jetty supports other key-store formats as well, for example BCFKS key format. BCFKS is FIPS compatible but the default JKS is not.
This way we will be able configure the key-store format from phoenix-queryserver.

Comment (joshelser) [reply server/src/main/java/org/apache/calcite/avatica/server/HttpServer.java:875]: thanks for the explanation.

**Comment by joshelser** [comment]
Please update the website with the new user-facing API and then I think this is 👍🏼

**Comment by F21** [comment]
Also, please squash your commits as well 😄

**Review by F21 — APPROVED** [review]
LGTM! Thanks @richardantal and @joshelser !

**Comment by richardantal** [comment]
Thank you for the reviews!

**Comment by joshelser** [comment]
Thanks Richie! Appreciate you merging, Francis!

**Pull request merged**